File: python/example/quicksort/quicksort.py
import sys
from example.quicksort.service.quicksort_service import QuicksortService
from otterop.lang.array import Array
from otterop.lang.string import String
from otterop.io.console import Console
println = Console.println

class Quicksort:

    @staticmethod
    def main(args):
        args_array = Array.wrap_string(args)
        QuicksortService.quick_sort(args_array, 0, args_array.size())
        i = 0
        while i < args_array.size():
            println(args_array.get(i))
            i = i + 1

if __name__ == "__main__":
    Quicksort.main(sys.argv[1:])

File: ts/example/quicksort/Quicksort.ts
import { QuicksortService } from './service/QuicksortService';
import { Array } from '@otterop/lang/array';
import { String } from '@otterop/lang/string';
import { Result } from '@otterop/lang/result';
import { Console } from '@otterop/io/console';
const { println } = Console;

export class Quicksort {

    public static main(args : string[]) : void {
        let argsArray = Array.wrapString(args);
        QuicksortService.quickSort(argsArray, 0, argsArray.size());
        let i = 0;
        
        while (i < argsArray.size()) {
            println(argsArray.get(i));
            i = i + 1;
        }
        Result.of(String.wrap(""), String.wrap(""));
    }
}

Quicksort.main(process.argv.slice(2));